src/Component/ProcessRunner/ProcessRunner.php

```diff
@@ -43,6 +43,7 @@ public function run(Host $host, string $command, array $config = []): string
             'idle_timeout' => null,
             'cwd' => defined('DEPLOYER_ROOT') ? DEPLOYER_ROOT : null,
             'tty' => false,
+            'vars' => [],
         ];
```

Comment:
wdyt about adding a phpdoc describing the $config array structure, so phpstan/psalm based workflows can validate it and catch errors eary?

something like
`@psalm-param array{timeout?: int, idle_timeout?: int|null, cwd?: string, tty?: bool, vars?: array<string, string>}`

?

Reply:
That's up to @antonmedv ;) At this point I personally would like to concentrate on this ticket and #2187 

If the project was to use the psalm annotations, then I guess it would be good to also include it in the pipeline and maybe have some kind of check, that makes sure that new commits include them

Reply:
added a phpstan github action with https://github.com/deployphp/deployer/pull/2218